inc/admin/network/class-sharingandprivacyoptions.php

```diff
@@ -207,6 +208,73 @@ public function updateBooksLastUpdatedDate() {
 		}
 	}
 
+	/**
+	 * Performs network book directory exclusion logic for non catalog books
+	 * @param bool $exclude   True for exclude and false for removing exclude
+	 */
+	public static function networkExcludeOption( bool $exclude ) {
+		if ( $exclude ) {
+			self::excludeNonCatalogBooksFromDirectory( 'excludeNonCatalogBooksFromDirectoryAction' );
+		} else {
+			self::excludeNonCatalogBooksFromDirectory( 'excludeNonCatalogBooksFromDirectoryAction', true );
+		}
+	}
+
+	/**
+	 * Triggers a batch book directory delete for all NON catalog books
+	 * @param bool $revert  un-checking network exclude
+	 */
+	static function excludeNonCatalogBooksFromDirectory( $callback, bool $revert = false ) {
+		$book_ids = self::getNonCatalogBooks();
+
+		if ( count( $book_ids ) > 0 ) {
+			self::$callback( $book_ids, $revert );
+		}
+	}
+
+	/**
+	 *  Returns all non catalog book ids
+	 * @return array    Non catalog books
+	 */
+	static function getNonCatalogBooks() {
+		$book_ids = [];
+		$books = get_sites( [ 'site__not_in' => [ 1 ] ] );
+		foreach ( $books as $book ) {
+			$in_catalog = get_site_meta( $book->blog_id, \Pressbooks\DataCollector\Book::IN_CATALOG, true );
+			if ( isset( $in_catalog ) && $in_catalog === '0' ) {
+				$book_ids[] = $book->blog_id;
+			}
+		}
+		return $book_ids;
+	}
+
+	/**
+	 * Perform actions during network book exclusion is enabled
+	 * @param $book_ids
+	 * @return array   Responses from actions
+	 */
+	static function excludeNonCatalogBooksFromDirectoryAction( array $book_ids, bool $revert = false ) {
+		$is_deleted = false;
+
+		if ( ! $revert ) {
+			$is_deleted = BookDirectory::init()->deleteBookFromDirectory( $book_ids );
+		}
+
+		$update_blogs = array_map(
```

Comment:
Before update last_updated field we should make sure that books were deleted.

Reply:
After consideration, I think it would be good to update the books independent of book fetcher's deletion process be cause in case of failure, the updated book may be caught from next sync. what do you think?

Reply:
Ho Man, as we discussed today could you create a separate ticket for this situation (linking it) please?
In that ticket would be nice research and solve:
- log those books that could not sync to fetcher for some reason
- create mocks for test this feature and change the logic to: update the blogs only if books were deleted from fetcher.